base/cvd/cuttlefish/host/commands/cvd/selector/instance_selector.cpp

```diff
@@ -22,46 +22,28 @@
 
 #include "host/commands/cvd/selector/device_selector_utils.h"
 #include "host/commands/cvd/selector/instance_database_types.h"
+#include "host/commands/cvd/selector/selector_common_parser.h"
 #include "host/commands/cvd/selector/selector_constants.h"
 #include "host/libs/config/config_constants.h"
 
 namespace cuttlefish {
 namespace selector {
 
 Result<InstanceSelector> InstanceSelector::GetSelector(
-    const cvd_common::Args& selector_args, const Queries& extra_queries,
+    const SelectorOptions& selector_options, const Queries& extra_queries,
     const cvd_common::Envs& envs) {
-  cvd_common::Args selector_args_copied{selector_args};
-  SelectorCommonParser common_parser =
-      CF_EXPECT(SelectorCommonParser::Parse(selector_args_copied, envs));
-  std::stringstream unused_args;
-  unused_args << "{";
-  for (const auto& arg : selector_args_copied) {
-    unused_args << arg << ", ";
-  }
-  std::string unused_arg_list = unused_args.str();
-  if (!selector_args_copied.empty()) {
-    unused_arg_list.pop_back();
-    unused_arg_list.pop_back();
-  }
-  unused_arg_list.append("}");
-  if (!selector_args_copied.empty()) {
-    LOG(ERROR) << "Warning: there are unused selector options. "
-               << unused_arg_list;
-  }
-
   // search by instance and instances
   // search by HOME if overridden
   Queries queries;
-  if (IsHomeOverridden(common_parser)) {
-    CF_EXPECT(common_parser.Home());
-    queries.emplace_back(kHomeField, common_parser.Home().value());
+  auto overriden_home_res = OverridenHomeDirectory(envs);
+  if (overriden_home_res.ok() && (*overriden_home_res).has_value()) {
+    queries.emplace_back(kHomeField, (*overriden_home_res).value());
   }
-  if (common_parser.GroupName()) {
-    queries.emplace_back(kGroupNameField, common_parser.GroupName().value());
+  if (selector_options.group_name) {
```

Comment:
Can this conversion be a member function of `CommandRequest`? It looks duplicated in `group_selector.cpp`.

Reply:
Done.